code-build.js

```diff
@@ -50,25 +53,47 @@ async function waitForBuildEndTime(sdk, { id, logs }, nextToken) {
       cloudWatchLogs
         .getLogEvents({ logGroupName, logStreamName, startFromHead, nextToken })
         .promise()
-  ]);
-  // Pluck off the relevant state
-  const [current] = batch.builds;
-  const { nextForwardToken, events = [] } = cloudWatch;
-
-  // stdout the CloudWatchLog (everyone likes progress...)
-  // CloudWatchLogs have line endings.
-  // I trim and then log each line
-  // to ensure that the line ending is OS specific.
-  events.forEach(({ message }) => console.log(message.trimEnd()));
-
-  // We did it! We can stop looking!
-  if (current.endTime && !events.length) return current;
-
-  // More to do: Sleep for 5 seconds :)
-  await new Promise(resolve => setTimeout(resolve, wait));
+  ]).catch((err) => {
+    errorDetected = true;
```

Comment:
I see what you are doing here but it will not work.
You are returning `undefined` which is not iterable.

Also, we are going to need a test :)

Reply:
I'm not sure what it should return in the case where this errors then.  If we reach the `catch`, don't we not really care about the return value because we're going to try again?

Reply:
if you do not at least return `[]` in the code as you have it,
the `catch` block will result in the `Promise.all(` resolving to `undefined`.
This is then spread to `const [batch, cloudWatch = {}] = await Promise.all([`

So to collapse the promise, in your current catch code you are effectively doing:
`const [batch, cloudWatch = {}] = undefined`
Which will throw `TypeError: undefined is not iterable`,
and you will never get to test your error :(

If you write a test for this,
like this: https://github.com/aws-actions/aws-codebuild-run-build/blob/master/test/code-build-test.js#L256
you can build up and `sdk` instance that will throw the error :)

Reply:
Yup, working on figuring out a way to get it to throw an error within the mock sdk.  Almost got it sorted out.